libs/core/src/lib/select/option/option.component.ts

```diff
@@ -63,97 +88,147 @@ export class OptionComponent implements OnInit, OnDestroy {
+    /**@hidden
+     *  Since select components could be using the option's label to display the selected values
+     *  and they don't have a way of knowing if the option's label has changed
+     * we have to check for changes in the DOM ourselves and dispatch an event
+     */
+    ngAfterViewChecked(): void {
+        if (this.selected) {
+            const viewValue = this.viewValue;
+
+            if (viewValue !== this._mostRecentViewValue) {
+                this._mostRecentViewValue = viewValue;
+                this._stateChanges.next();
+            }
+        }
     }
 
     /** @hidden */
     ngOnDestroy(): void {
-        this._subscriptions.unsubscribe();
+        this._stateChanges.complete();
     }
 
-    /** Returns the view value text of the option, or the viewValue input if it exists. */
-    get viewValueText(): string {
-        return (this._elementRef.nativeElement.textContent || '').trim();
+    /** @hidden
+ * This method sets display styles on the option to make it appear
+ * active. This is used by the ActiveDescendantKeyManager so key
+ * events will display the proper options as active on arrow key events.
+ */
+    setActiveStyles(): void {
+        if (!this._active) {
+            this._active = true;
+            this._changeDetectorRef.markForCheck();
+        }
     }
 
-    /** Set control selected state
-     * @param value - whether is selected
-     * @param controlChange - whether is a change initially coming from OptionComponent
-     * */
-    setSelected(value: boolean, controlChange: boolean): void {
-        this.selected = value;
+    /** @hidden
+     * This method removes display styles on the option that made it appear
+     * active. This is used by the ActiveDescendantKeyManager so key
+     * events will display the proper options as active on arrow key events.
+     */
+    setInactiveStyles(): void {
+        if (this._active) {
+            this._active = false;
+            this._changeDetectorRef.markForCheck();
+        }
+    }
+
+    /** @hidden
+     * Focuses the element. */
+    focus(_origin?: FocusOrigin, options?: FocusOptions): void {
+
+        const element = this._elementRef.nativeElement;
 
-        if (value) {
-            this._selectProxy.optionStateChange$.next({option: this, controlChange: controlChange})
+        if (typeof element.focus === 'function') {
+            element.focus(options);
         }
-        this._changeDetRef.markForCheck();
     }
 
-    /** Focuses the element. */
-    focus(): void {
-        (this._elementRef.nativeElement as HTMLElement).focus();
+    /** @hidden
+   * option value
+   */
+    getLabel(): string {
+        return this.viewValue;
     }
 
-    /** Returns HTMLElement representation of the component. */
-    getHtmlElement(): HTMLElement {
+    /** @hidden
+     * Returns HTMLElement representation of the component. */
+    _getHtmlElement(): HTMLElement {
         return this._elementRef.nativeElement as HTMLElement;
     }
```

Comment:
Looks like unused method

Reply:
So it's used in specs, got it.
In this case no need for `as` cause _elementRef already has type reference